docs/telemetry-perf.md

```diff
@@ -258,3 +258,66 @@ How long it took from when the user stopped pressing a key to when they were sho
         rHandler->>User: add to already shown results
         end
 ```
+
+## Crash Monitoring
+
+We make an attempt to gather information regarding when the IDE crashes. This is the diagram of the steps that take place.
+
+## Sequence Diagram
+
+`Crash Monitoring`: The "service" that allows us to detect and react to crashes
+
+`Heartbeat`: Sends heartbeats which signal that the extension is still running and has not crashed
+
+`Crash Checker`: Observes the heartbeats, reporting a telemetry event if a crash is detected
+
+`File System State`: The user's file system that we store our heartbeat state
+
```

Comment:
markdown supports lists, let's use them

Reply:
I know that, this was a stylistic choice